core/src/com/unciv/models/translations/Translations.kt

```diff
@@ -213,6 +241,57 @@ fun String.tr(): String {
     val activeMods = with(UncivGame.Current) {
         if (isGameInfoInitialized()) gameInfo.gameParameters.mods else translations.translationActiveMods
     }
+    val language = UncivGame.Current.settings.language
+    
+    if (contains('<')) { // Conditionals!
+        /**
+         * So conditionals can contain placeholders, such as <vs [unitFilter] units>, which themselves
+         * can contain multiple filters, such as <vs [{Military} {Water}] units>.
+         * Moreover, we can have any amount of conditionals in any order, and translations
+         * can reorder these conditionals in any way they like, even putting them in front
+         * of the rest of the translatable string.
+         * All of this nesting makes it quite difficult to translate, and is the reason we check
+         * for these first.
+         * 
+         * The plan: First translate each of the conditionals on its own, and then combine them
+         * together into the final fully translated string.
+         */
+        
+        val baseText = this.removeConditionals()
+        val translatedBaseText = baseText.tr()
+        
+        val conditionals = this.getConditionals().map { it.placeholderText }
+        val conditionsWithTranslation: HashMap<String, String> = hashMapOf()
+        
+        for (conditional in this.getConditionals())
+            conditionsWithTranslation[conditional.placeholderText] = conditional.text.tr()
+        
+        val translatedConditionals: MutableList<String> = mutableListOf()
+        
+        // Somewhere, we asked the translators to reorder all possible conditionals in a way that
+        // makes sense in their language. We get this ordering, and than extract each of the
+        // translated conditionals, removing the <> surrounding them, and removing placeholder text
+        // where if it exists.
+        val conditionalOrdering = UncivGame.Current.translations.getConditionalOrder(language)
+        for (placedConditional in pointyBraceRegex.findAll(conditionalOrdering).map { it.value.substring(1, it.value.length-1).getPlaceholderText() }.toList()) {
```

Comment:
Hmmm, Without really checking - the toList should be redundant, you're not modifying anything the iterator sees. And removing first and last char should be possible to already do in the regex capturing - looks like it already does. The capture won't contain the braces. So don't use the entire match, use the capture. Or if you don't use the capture, don't capture.